velox/exec/tests/ContainerRowSerdeTest.cpp

```diff
@@ -144,5 +186,126 @@ TEST_F(ContainerRowSerdeTest, nested) {
   testRoundTrip(nestedArray);
 }
 
+TEST_F(ContainerRowSerdeTest, compareNullsInArrayVector) {
+  auto data = makeArrayVector<int64_t>({
+      {1, 2},
+      {1, 5},
+      {1, 3, 5},
+      {1, 2, 3, 4},
+  });
+
+  std::vector<HashStringAllocator::Position> positions =
+      serializeWithPositions(data);
+  auto arrayVector = makeNullableArrayVector<int64_t>({
+      {1, 2},
+      {1, 3},
+      {1, 5},
+      {std::nullopt, 1},
+  });
+  SelectivityVector rows(arrayVector->size());
+  DecodedVector decodedVector;
+  decodedVector.decode(*arrayVector, rows);
+
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{0}, {1}, {-1}, std::nullopt},
+      false,
+      CompareFlags::NullHandlingMode::StopAtNull);
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{0}, {1}, {1}, {1}},
+      true,
+      CompareFlags::NullHandlingMode::StopAtNull);
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{0}, {1}, {-1}, {1}},
+      false,
+      CompareFlags::NullHandlingMode::NoStop);
+
+  allocator_.clear();
+}
+
+TEST_F(ContainerRowSerdeTest, compareNullsInMapVector) {
+  auto data = makeNullableMapVector<int64_t, int64_t>({
+      {{{1, 10}, {4, 30}, {2, 3}}},
+      {{{2, 20}}},
+      {{{3, 50}}},
+      {{{4, std::nullopt}}},
+  });
+  std::vector<HashStringAllocator::Position> positions =
+      serializeWithPositions(data);
+  auto mapVector = makeNullableMapVector<int64_t, int64_t>({
+      {{{1, 10}, {3, 20}}},
+      {{{2, 20}}},
+      {{{3, 40}}},
+      {{{4, std::nullopt}}},
+  });
+
+  SelectivityVector rows(mapVector->size());
+  DecodedVector decodedVector;
+  decodedVector.decode(*mapVector, rows);
+
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{-1}, {0}, {1}, std::nullopt},
+      false,
+      CompareFlags::NullHandlingMode::StopAtNull);
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{1}, {0}, {1}, std::nullopt},
+      true,
+      CompareFlags::NullHandlingMode::StopAtNull);
+  testCompareWithNulls(
+      decodedVector,
+      positions,
+      {{1}, {0}, {1}, {0}},
+      true,
+      CompareFlags::NullHandlingMode::NoStop);
+
+  allocator_.clear();
+}
+
+TEST_F(ContainerRowSerdeTest, compareNullsInRowVector) {
+  auto data = makeRowVector({makeFlatVector<int32_t>({
+      1,
+      2,
+      3,
+      4,
+  })});
+  std::vector<HashStringAllocator::Position> positions =
+      serializeWithPositions(data);
+
+  auto someNulls = makeNullableFlatVector<int32_t>({
+      {1},
+      {3},
+      {2},
+      std::nullopt,
+  });
+  auto rowVector = makeRowVector({someNulls});
+  SelectivityVector rows(rowVector->size());
+  DecodedVector decodedVector;
+  decodedVector.decode(*rowVector, rows);
```

Comment:
You can remove 'rows' parameter and then remove 'rows' variable.

DecodedVector decodedVector(*decodedVector);

Reply:
Fixed